src/containers/tutor-home-page/subjects-step/SubjectsStep.jsx

```diff
@@ -1,13 +1,106 @@
+import { useEffect, useState } from 'react'
 import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import { useTranslation } from 'react-i18next'
 
+import AppAutoComplete from '~/components/app-auto-complete/AppAutoComplete'
+import { useStepContext } from '~/context/step-context'
+import { categoryService } from '~/services/category-service'
+import { subjectService } from '~/services/subject-service'
+
+import img from '~/assets/img/tutor-home-page/become-tutor/study-category.svg'
 import { styles } from '~/containers/tutor-home-page/subjects-step/SubjectsStep.styles'
 
+const STEP_KEY = 'subjects'
+
 const SubjectsStep = ({ btnsBox }) => {
+  const { t } = useTranslation()
+  const { stepData, handleStepData } = useStepContext()
+
+  const ctx = stepData?.[STEP_KEY]
+  const data = ctx?.data ?? { category: null, subjects: null }
+  const errors = ctx?.errors ?? {}
+
+  useEffect(() => {
+    if (!ctx) handleStepData(STEP_KEY, data, errors)
+  }, [])
```

Comment:
Think about reorganizing this, add dependencies to `useEffect` 
```suggestion
  const ctx = stepData?.[STEP_KEY]
  useEffect(() => {
    const data = ctx?.data ?? { category: null, subjects: null }
    const errors = ctx?.errors ?? {}
    if (!ctx) handleStepData(STEP_KEY, data, errors)
  }, [ctx ])
```